drivers/MbedCRC.h

```diff
@@ -0,0 +1,29 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_CRC_API_H
+#define MBED_CRC_API_H
+
+#include <stdint.h>
+
+#if DEVICE_CRC
```

Comment:
I would like to leave it default this way, and create class later on (based on HW), as it might result in lot more changes later (specially in constructor) if we have different polynomials supported by different HW.

Reply:
That's also fine.

Reply:
Not sure why we would leave this condition here. The class does not exist, neither device_has CRC. Once it is defined, this would be added here. Otherwise it is dead code, if I define CRC in `device_has` or via command line, it does not build.

Reply:
Agree @0xc0170  Entire device CRC code below is dead code, but it was kept to add default CRC as HW (which is not present now) and switch back to FastCRC as default.
`+#if DEVICE_CRC
 +#include "HwCRC.h"
 +typedef mbed::HwCRC RawCRC;`

I can either added error here (#error HwCRC not supported), or leave this code commented.